src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,289 @@
+        /// <returns>A 10 bit value corresponding to relative voltage level on specified device channel</returns>
+        public int ReadChannel(Channel channelNumber)
+        {
+            Read((int)channelNumber);
+            return Read((int)ChargeChannel);
+        }
+
+        /// <summary>
+        /// Reads sensor values into an List
+        /// </summary>
+        /// <param name="channelList">List of channels to read</param>
+        /// <returns>List of 10 bit values corresponding to relative voltage level on specified device channels</returns>
+        public List<int> ReadChannels(List<Channel> channelList)
+        {
+            List<int> values = new List<int>(channelList.Count);
+            Read((int)channelList[0]);
+            for (int i = 1; i < channelList.Count; i++)
+            {
+                values.Add(Read((int)channelList[i]));
+            }
+
+            values.Add(Read((int)ChargeChannel));
+            return values;
+        }
+
+        /// <summary>
+        /// Reads all sensor values into an List
+        /// </summary>
+        /// <returns>List of 10 bit values corresponding to relative voltage level on all channels</returns>
+        public List<int> ReadAll()
+        {
+            List<int> values = new List<int>(_channels);
+            Read(0);
+            for (int i = 1; i < values.Count; i++)
+            {
+                values.Add(Read(i));
+            }
+
+            values.Add(Read((int)ChargeChannel));
+            return values;
+        }
+
+        /// <summary>
+        /// Mode 1 (Fast mode):
+        /// <br>10 clock transfer with !CS high </br>
+        /// <br>Figure 9 in datasheet</br>
+        /// </summary>
+        /// <param name="channelNumber">Channel to read</param>
+        /// <returns>10 bit value corresponding to relative voltage level on channel</returns>
+        public int Read(int channelNumber)
+        {
+            int value = 0;
+            _digital.Write(_chipSelect, 0);
+            DelayHelper.DelayMicroseconds(1, false);
+            for (int i = 0; i < 10; i++)
+            {
+                if (i < 4)
+                {
+                    // send 4-bit Address
+                    if ((channelNumber >> (3 - i) & 0x01) != 0)
+                    {
+                        _digital.Write(_address, 1);
+                    }
+                    else
+                    {
+                        _digital.Write(_address, 0);
+                    }
+                }
+
+                // read 10-bit data
+                value <<= 1;
+                if (_digital.Read(_dataOut) == PinValue.High)
+                {
+                    value |= 0x01;
+                }
+
+                _digital.Write(_inputOutputClock, 1);
+                _digital.Write(_inputOutputClock, 0);
+            }
+
+            _digital.Write(_chipSelect, 1);
+            DelayHelper.DelayMicroseconds(100, true); // t(PZH/PZL)

```

Comment:
please place the comment up the line and please refer as well to the documentation for the delay